src/Symfony/Component/Routing/Generator/UrlGenerator.php

```diff
@@ -295,6 +295,12 @@ protected function doGenerate(array $variables, array $defaults, array $requirem
             return $a == $b ? 0 : 1;
         });
 
+        array_walk_recursive($extra, static function (&$value) {
+            if (null !== $value) {
```

Comment:
let's make this a bit safer?
```php

array_walk_recursive($extra, $caster = \Closure::bind(static function (&$v) use (&$caster) {
    if (\is_object($v) && $p = get_object_vars($v)) {
        array_walk_recursive($p, $caster);
		$v = $p;
    } elseif (null !== $v) {
	    $v = (string) $v;
    }
}, null, null));
```

Reply:
I think I'd prefer this actually :)
```
array_walk_recursive($extra, $caster = \Closure::bind(static function (&$v) use (&$caster) {
    if ($v instanceof \stdClass) {
		$v = (array) $v;
        array_walk_recursive($v, $caster);
    } elseif (null !== $v) {
	    $v = (string) $v;
    }
}, null, null));
```

Reply:
Done

Reply:
BTw, is `use (&$caster)` really needed? Tests are running without it 🤔

Reply:
Added